Source/drlg_l1.cpp

```diff
@@ -1950,17 +1950,21 @@ void drlg_l1_crypt_rndset(const BYTE *miniset, int rndper)
 			int kk = sw * sh + 2;
 			if (miniset[kk] >= 84 && miniset[kk] <= 100 && found) {
 				// BUGFIX: accesses to dungeon can go out of bounds (fixed)
-				// BUGFIX: Comparisons vs 100 should use same tile as comparisons vs 84 (fixed)
-				if (sx > 0 && dungeon[sx - 1][sy] >= 84 && dungeon[sx - 1][sy] <= 100) {
+				// BUGFIX: Comparisons vs 100 should use same tile as comparisons vs 84 - NOT A BUG - "fixing" this breaks crypt
+
+				auto comparison_with_bounds_check = [](Point p1, Point p2) {
+					return (p1.x >= 0 && p1.x < DMAXX && p1.y >= 0 && p1.y < DMAXY) && (p2.x >= 0 && p2.x < DMAXX && p2.y >= 0 && p2.y < DMAXY) && (dungeon[p1.x][p1.y] >= 84 && dungeon[p2.x][p2.y] <= 100);
+				};

```

Comment:
How about splitting this as a separate function instead of a lambda, doesn't seam that we really get any benefits from it being a lambda.

Reply:
The lambda is less visible than a private method, so it is preferred if no one needs access to it outside the parent function IMHO.